src/components/PostsList.tsx

```diff
@@ -1,54 +1,54 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
 
 import classNames from 'classnames';
-import React from 'react';
-import { Post } from '../types/Post';
+import { FC } from 'react';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { setSelectedPost } from '../features/selectedPost';
 
-type Props = {
-  posts: Post[];
-  selectedPostId?: number;
-  onPostSelected: (post: Post | null) => void;
-};
+export const PostsList: FC = () => {
+  const posts = useAppSelector(state => state.posts.posts);
 
-export const PostsList: React.FC<Props> = ({
-  posts,
-  selectedPostId = 0,
-  onPostSelected,
-}) => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+  const selectedPostId = useAppSelector(state => state.selectedPost?.id);
```

Comment:
Same as above, consider creating a custom hook for fetching the selected post ID. This will allow you to encapsulate the logic of fetching the selected post ID, making the code cleaner and easier to test.